ifttt/watches/gcloud_datastore_aggregation.py

```diff
@@ -19,17 +22,27 @@ def __repr__(self):
         return "AggregatedDatastoreWatch '{}'".format(self.name)
 
     def collect_activations(self):
-        collector = list()
+        collector = collections.defaultdict(list)
 
         query = self.client.query(kind=self.kind)
         for result in query.fetch():
             value = result.get(self.field)
             if not value:
                 continue  # TODO: aggregates across nulls
 
-            collector.append(value)
+            context = None
+            if self.context_field:
+                context = result.get(self.context_field)
+                context = context or None  # TODO: consider handling nulls vs ''
 
-        prev = self.cache[self.cache_key].get(self.field, 0)  # TODO: aggregates across nulls
-        curr = self.aggregate_fn(collector)
-        if self.if_fn(self.cache_key, prev, curr):
-            yield self.cache_key, curr
+            collector[context].append(value)
+
+        for context, collection in collector.items():
+            cache_key = self.cache_key
+            if context:
+                cache_key += '-{}'.format(context)
+
+            prev = self.cache[cache_key].get(self.field, 0)  # TODO: aggregates across nulls
+            curr = self.aggregate_fn(collection)
+            if self.if_fn(self.cache_key, prev, curr):
+                yield self.cache_key, context, curr
```

Comment:
This is for sure supposed to be `self.cache_key` and not the specific `cache_key` which this iteration is returning?

Reply:
It is indeed. I should make this more clear.

Reply:
On second thought, it could go either way. Adding a comment explaining reasoning and updating the usage.